### DISCUSSION TITLE
안녕하십니까 작가님 질문이 있습니다

### DISCUSSION THREAD

**Discussion by Onthesly** [fsoftwareengineer/todo-application#23] [질문과 답변] [answered]
저는 프로그래밍 초보입니다.
스프링MVC 모델만 배우다가 그래도 최신기술인 리액트, 스프링부트를 배우고싶어서 책을 구입하고 따라서 프로젝트를 진행하고 있습니다.
책이 잘 구성되어있어 아직까지 막힘없이 배우고 있습니다. 감사합니다.

다름이 아니라 요번에 log4j 라이브러리 취약점 사태가 일어나서 관련자료를 보고 있습니다.
질문은 책에서 하는 프로젝트에서는 slf4j를 사용해서 관련이 없는지 궁금합니다.
최대한 혼자 알아보려고했는데 기본기가 부족하다보니 이해가 잘되지 않습니다.(spring-boot-starter-web에는 기본적으로 logback을 쓴다고도 하고..slf4j는 자체적인 로깅 프레임워크가 아니라 다른 로깅 프레임워크가 접근할수 있도록 도와주는 추상화계층이라고도 하고..)
제 project dependency에는 log4j, log4j-to-slf4j, logback-classic&core, slf4j-api등 여러가지가 혼재되어있습니다.

마지막까지해서 AWS를 이용해 배포하고자 하는 목표를 가지고있어서 염려가 되어 여쭙습니다.
베이스 지식이 없이 질문을 드리는것 같아서 죄송합니다..ㅠ

**Comment by fsoftwareengineer** [comment, marked as answer]
스프링 부트에서 로깅방법을 log4j로 변경하지 않은 경우에는 괜찮다고 합니다. 하지만 확실히 하기 위해서 log4j버전을 다음과같이 지정 할 수 있습니다.

build.gradle
ext['log4j2.version'] = '2.15.0'

설정 후 아래와 같은 명령으로 log4j-core 버전을 확인 할 수 있습니다.

./gradlew dependencyInsight --dependency log4j-core


이와 관련해서는 따로 공지를 올리도록 하겠습니다.

**Reply by fsoftwareengineer** [reply]
참고: https://spring.io/blog/2021/12/10/log4j2-vulnerability-and-spring-boot

**Reply by Onthesly** [reply]
제 프로젝트 dependency에 log4j-core는 없네요! 답변과 링크 정말 감사합니다!!
